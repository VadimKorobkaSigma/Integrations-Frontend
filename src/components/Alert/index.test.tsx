/* eslint-disable @typescript-eslint/no-explicit-any */
import * as React from 'react';
import Alert from './index';
import { shallow } from 'enzyme';

test('Alert Component with children', () => {
    const props = {
        options: {
            type: '',
        },
        message: 'Test message',
        close: jest.fn(),
    } as any;
    const wrapper = shallow(<Alert {...props}>Submit</Alert>);
    expect(wrapper.html()).toMatchSnapshot();

    for (const type of ['success', 'info', 'error']) {
        props.options.type = type;
        wrapper.setProps({ ...props, options: { ...props.options } });
        expect(wrapper.html()).toMatchSnapshot();
    }

    wrapper.find('div').simulate('click');
    expect(props.close).toBeCalledTimes(1);
});
