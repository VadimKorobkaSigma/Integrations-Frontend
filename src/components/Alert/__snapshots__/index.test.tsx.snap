// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component with children 1`] = `"<div class=\\"\\"><p>Test message</p><img alt=\\"close icon\\" data-name=\\"close-button\\"/></div>"`;

exports[`Alert Component with children 2`] = `"<div class=\\"\\"><img alt=\\"success icon\\"/><p>Test message</p><img alt=\\"close icon\\" data-name=\\"close-button\\"/></div>"`;

exports[`Alert Component with children 3`] = `"<div class=\\"\\">!<p>Test message</p><img alt=\\"close icon\\" data-name=\\"close-button\\"/></div>"`;

exports[`Alert Component with children 4`] = `"<div class=\\"\\"><img alt=\\"error icon\\"/><p>Test message</p><img alt=\\"close icon\\" data-name=\\"close-button\\"/></div>"`;
